### PR TITLE
sys_fs hotfix

### PR DESCRIPTION
**Pull request by elad335**
Attempt to address #13009 (ENOTMOUNTED errors were inactive due to previous change)

### CODE REVIEW

**Comment by Megamouse** [comment]
i don't know if you noticed but there is literally a PR open that touches the same code

**Comment by elad335** [comment]
Right.

**Comment by elad335** [comment]
Fix is too urgent, integrated his changes and added signature.

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp [hunk 1]
```diff
@@ -206,31 +207,39 @@ std::string_view lv2_fs_object::get_device_path(std::string_view filename)
...
+			if (mp_root == mp_name)
+			{
+				return mp;
+			}
+		}
+	}
 
 	// Default fallback
-	return &g_mp_sys_dev_root;
+	return &g_mp_sys_dev_hdd0;
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_fs.cpp:242]: the default fallback used to be g_mp_sys_no_device

**Comment by brian218** [comment]
Pardon, do we really need `g_mp_sys_no_device`? It is not used anywhere else.

**Comment by elad335** [comment]
Yes, it fixes a related issue we are trying to solve which is wrong pointer equalities leading to wrong error checks.

**Comment by elad335** [comment]
@brian218 https://github.com/RPCS3/rpcs3/pull/9027

**Comment by brian218** [comment]
> @brian218 #9027

I wonder why `g_mp_sys_dev_hdd0` can't be used in this case.

**Comment by elad335** [comment]
It can, but for now, let's make == &g_sys_dev_hdd0 comparisons fail with relative paths.

**Pull request merged**